x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ExponentialHistogramArrayBlock.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.ReleasableIterator;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.exponentialhistogram.AbstractExponentialHistogram;
+import org.elasticsearch.exponentialhistogram.ExponentialHistogram;
+import org.elasticsearch.exponentialhistogram.ExponentialHistogramCircuitBreaker;
+import org.elasticsearch.exponentialhistogram.ZeroBucket;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public final class ExponentialHistogramArrayBlock extends AbstractNonThreadSafeRefCounted implements ExponentialHistogramBlock {
+
+    // TODO(b/133393): Add blocks for min/max/sum/count
+    private final BytesRefBlock encodedHistograms;
+    private boolean isClosed = false;
+
+    public ExponentialHistogramArrayBlock(BytesRefBlock encodedHistograms) {
+        this.encodedHistograms = encodedHistograms;
+    }
+
+    @Override
+    public ExponentialHistogram getExponentialHistogram(int valueIndex) {
```

Comment:
Can we return the encoded histogram and let callers decode and use it - we also need to pass the scratch parameter. We can add a helper method in this class for decoding.

Reply:
That's kind of what I inteded to do - but abstracted away.
The key here is that the `ExponentialHistogram` intentionally does not allow random access on the buckets. You can only iterate over them.
This allows us to do encoding lazily in the `ExponentialHistogram` implementation we return here without having to repeat this on every consumer of this block.

So for that reason I don't see the benefit of your suggestion here.